pennylane/ops/qubit/special_unitary.py

```diff
@@ -125,6 +125,27 @@ def pauli_basis_strings(num_wires):
     return ["".join(letters) for letters in product(_pauli_letters, repeat=num_wires)][1:]
 
 
+def _pauli_decompose(matrix, num_wires):
+    """Compute the coefficients of a matrix or a batch of matrices (batch dimension(s) in the
+    leading axes) in the Pauli basis.
```

Comment:
Can you add here that what this is explicitly computing is the $c_i$ coefficients in $M = \sum_i c_i P_i$ for Pauli basis $\{P_i\}$ via $c_i = \text{tr}(P_i M)$ - and maybe a developer comment in the code that this done in a vectorized way here 👍